compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -820,6 +831,96 @@ public void visit(BLangIntRangeExpression intRangeExpression) {
         resultTypes = Lists.of(new BArrayType(symTable.intType));
     }
 
+    @Override
+    public void visit(BLangTableQueryExpression tableQueryExpression) {
+        BType actualType = symTable.errType;
+        int expTypeTag = expTypes.get(0).tag;
+
+        if (expTypeTag == TypeTags.TABLE) {
+            actualType = expTypes.get(0);
+        } else if (expTypeTag != TypeTags.ERROR) {
+            dlog.error(tableQueryExpression.pos, DiagnosticCode.INCOMPATIBLE_TYPES_CONVERSION, expTypes.get(0));
+        }
+
+        BLangTableQuery tableQuery = (BLangTableQuery) tableQueryExpression.getTableQuery();
+        tableQuery.accept(this);
+
+        resultTypes = types.checkTypes(tableQueryExpression, Lists.of(actualType), expTypes);
+    }
+
+    @Override
+    public void visit(BLangTableQuery tableQuery) {
+        BLangStreamingInput streamingInput = (BLangStreamingInput) tableQuery.getStreamingInput();
+        streamingInput.accept(this);
+
+        BLangJoinStreamingInput joinStreamingInput = (BLangJoinStreamingInput) tableQuery.getJoinStreamingInput();
+        if (joinStreamingInput != null) {
+            joinStreamingInput.accept(this);
+        }
+
+        /*
+        BLangSelectClause selectClause = (BLangSelectClause) tableQuery.getSelectClauseNode();
+        selectClause.accept(this);
+
+        BLangOrderBy orderBy = (BLangOrderBy) tableQuery.getOrderByNode();
+        if (orderBy != null) {
+            orderBy.accept(this);
+        }
+        */
```

Comment:
Shall we remove this commented code block?

Reply:
removed.